interactive_engine/executor/ir/core/src/glogue/pattern.rs

```diff
@@ -533,147 +571,117 @@ fn build_logical_plan(
     origin_pattern: &Pattern, mut exact_extend_steps: Vec<ExactExtendStep>,
 ) -> IrPatternResult<pb::LogicalPlan> {
     let mut match_plan = pb::LogicalPlan::default();
-    let mut child_offset: i32 = 1;
-    let source_extend = match exact_extend_steps.pop() {
-        Some(src_extend) => src_extend,
-        None => {
-            return Err(IrPatternError::InvalidExtendPattern(
-                "Build logical plan error: from empty extend steps!".to_string(),
-            ))
+    let source_extend = exact_extend_steps
+        .pop()
+        .ok_or(IrPatternError::InvalidExtendPattern(
+            "Build logical plan error: from empty extend steps!".to_string(),
+        ))?;
+    append_opr(&mut match_plan, generate_source_operator(origin_pattern, &source_extend)?)?;
+    for exact_extend_step in exact_extend_steps.into_iter().rev() {
+        let edge_expands_num = exact_extend_step.len();
+        // store all the expand operators in a 2D vector
+        // each 1D vector represent the expand operators correspond to a expand operator
+        let expand_oprs_vec =
+            exact_extend_step.generate_expand_operators_vec(origin_pattern, edge_expands_num > 1)?;
+        // store the current length of the match plan
+        let child_offset = match_plan.nodes.len();
+        // record the opr ids that are pre_node's children
+        let expand_chidren = get_expand_children(&expand_oprs_vec, child_offset);
+        // record the opr ids that are intersect's parents
+        let intersect_parents = get_intersect_parents(&expand_oprs_vec, child_offset);
+        // the id of the intersect operator
+        let intersect_id = get_intersect_id(&expand_oprs_vec, child_offset);
```

Comment:
compute the intersect related info when it is intersect case. i.e.,
 ```
// the case of intersect
if edge_expands_num > 1 {
     // move here.
}
```

Reply:
Fixed, but lead to the clone of expand_oprs_vec